### ISSUE TITLE
Support POST call to get next page for Paging API

### ISSUE THREAD

**Issue by lirenhe** [Azure/autorest#4454]
**Is your feature request related to a problem? Please describe.**
When working on data plane codegen for Metrics Adviser, there is a new pattern of use POST call to get next page.
However, according to https://azure.github.io/autorest/extensions/#x-ms-pageable,  the request to get next page is always a 'GET' call.

**Describe the solution you'd like**
Extend the definition in x-ms-pageable to support both GET and POST.

**Describe alternatives you've considered**
Need to write complex directive to workaround this problem which i don't think this is a good solution.

**Additional context**
N/A

**Comment by mikekistler** [comment]
It looks like the `x-ms-pageable` extension defines an `operationName` property that might be sufficient to satisfy this requirement.  If `operationName` is the `operationId` of a `POST`, then autorest could determine that it should generate a POST rather than GET.

**Comment by timotheeguerin** [comment]
@mikekistler @lirenhe is this then an issue? should the doc just be updated?

**Comment by mikekistler** [comment]
@timotheeguerin I guess what I was hoping is that you or someone could confirm that autorest __does__ generate a POST rather than a GET if `operationId` is the id of a POST.  (I should have made that explicit -- my bad)

**Comment by timotheeguerin** [comment]
tested that operationName pointing to an operation that is a post and it does look like to be respecting this in the generated code.

**Comment by mikekistler** [comment]
Sounds like this is already supported then. @lirenhe anything else you were looking for, or can we close this one?

**Comment by lirenhe** [comment]
This is not true for JS, even the operation is post, the call for next link is still 'GET'  https://github.com/AzureSDKAutomation/azure-sdk-for-js/pull/7966/files#diff-1f5caf1baebd30e4926a80f8447f674f99ffdf9d120f3e2d3cffa97211c7015bR593-R621

**Comment by timotheeguerin** [comment]
Do you have the `operationName` property set in the `x-ms-pageable`? I don't see it in the linked pr. This should be the same value of the operationId that is following.

**Comment by lirenhe** [comment]
How would operation name change the behaviour of POST/GET?
It is just string with some default value  (Optional (default: <operationName>Next). Specifies the name of the operation for retrieving the next page.)

You could check the comment in this PR  https://github.com/Azure/azure-rest-api-specs/pull/18123/files#r825306574

**Comment by mikekistler** [comment]
As I mentioned above,
> If `operationName` is the `operationId` of a POST, then autorest could (and Timothee confirmed _does_) determine that it should generate a POST rather than GET.

For the API definition you linked, you need to add:
```
    "operationName": "UserAssignedIdentities_ListAssociatedResources"
```
in the `x-ms-pageable` extension [at line 230](https://github.com/Azure/azure-rest-api-specs/pull/18123/files#diff-cde592398a0cad523af6c6822d19a93ee9de47f8f74449e95a6e6936d366a7b7R230).

**Comment by lirenhe** [comment]
@timotheeguerin   @mikekistler   thanks for your answer.
I would like to clarify a couple of behaviors for x-ms-pageable.

1. For the above sample, it seems both 'nextLinkName' and 'operationName' are defined.  Then which one shall we use to decide the URL to get the next batch?   I see dotnet use a combination of 'nextLinkName' and 'operationName', is this the right behavior?
      x-ms-pageable:
        nextLinkName: nextLink
        itemName: values
        operationName: Paging_nextOperationWithQueryParams

2. Usually, the API to get/post next batch of result is private that service team may not want to expose it in Swagger. Invoking this API directly will be rejected and there is no need to document this API.  
In above example, how can we define this /paging/multiple/nextOperationWithQueryParams as private API?

3.  From https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#collections, does it mean the value for 'nextLinkName' could be an absolute URL instead of a relative one?   So if that URL does not use the same endpoint from the original call, how can we workaround the origin problem by using 'operationName' instead?

**Comment by mikekistler** [comment]
My answers:
1. I believe both fields are necessary, since `nextLink` will contain specific parameter values that must be used to call `operationName`.
2. I think that the common scenario for pageable operations is that the API is "public" and already documented in the API definition file, and often is exactly the same operation that initiates the list operation, just using an undocumented parameter (often "continuationToken"). This isn't currently documented in the API definition because OpenAPI v2 has no way to define it.  It can be defined in OpenAPI v3.  If you know of different cases please share.
3. The guidelines say that `nextLink` should be an absolute URL. I know of no case where the endpoint for this URL is actually different, but if you do please share.

**Comment by lirenhe** [comment]
As this approach could work for most of the SDKs, i am good to leverage that to solve the problem of handling post for nextlink.
The only suggestion is that we could improve the doc https://azure.github.io/autorest/extensions/#x-ms-pageable otherwise service team does not know how to provide the correct swagger spec to handle this problem.
